tests/verify_heap.cpp

```diff
@@ -426,10 +430,13 @@ __global__ void deallocAll(
  *
  * @param data the datastructure to damage
  */
-__global__ void damageElement(allocElem_t ** data)
+struct DamageElement
```

Comment:
By any chance, can I just define the kernel as a normal function template? I could not get it compiling. Only a struct or a lambda seem to work with alpaka.

Reply:
Alpaka always gets a kernel is a functor. You can still have your implementation in a free function style, like in CUDA. Then just wrap it into a functor (lambda or `std::function`) when you create a kernel object instance to give to alpaka.

Reply:
Ok, thx for the explanation!

Reply:
`std::function` should not work on device.